server/service/device_client.go

```diff
@@ -0,0 +1,56 @@
+package service
+
+import (
+	"bytes"
+	"fmt"
+	"net/http"
+
+	"github.com/fleetdm/fleet/v4/server/fleet"
+)
+
+type DeviceClient struct {
+	*baseClient
+}
+
+func (dc *DeviceClient) request(verb string, path string, query string, responseDest interface{}) error {
+	var bodyBytes []byte
+	request, err := http.NewRequest(
+		verb,
+		dc.url(path, query).String(),
+		bytes.NewBuffer(bodyBytes),
+	)
+
+	if err != nil {
+		return err
+	}
+
+	response, err := dc.http.Do(request)
+	if err != nil {
+		return fmt.Errorf("%s %s: %w", verb, path, err)
+	}
+	defer response.Body.Close()
+
+	return dc.parseResponse(verb, path, response, responseDest)
+}
+
+// NewDeviceClient instantiates a new client to perform requests against device endpoints
+func NewDeviceClient(addr string, insecureSkipVerify bool, rootCA string) (*DeviceClient, error) {
+	baseClient, err := newBaseClient(addr, insecureSkipVerify, rootCA, "")
+
+	if err != nil {
+		return nil, err
+	}
+
+	return &DeviceClient{baseClient: baseClient}, nil
+}
+
+// ListDevicePolicies fetches all policies for the device with the provided token
+func (dc *DeviceClient) ListDevicePolicies(token string) ([]*fleet.HostPolicy, error) {
```

Comment:
I believe DeviceClient's public methods should have a ctx context.Context as first argument to allow for some control (timeout, cancellation, etc.).

Reply:
do you envision using that context soon? if not I'd rather wait until we really need it.

Reply:
+1, agree :)